File: 0206/main.swift
public class ListNode {
    public var val: Int
    public var next: ListNode?
    public init(_ val: Int) {
        self.val = val
        self.next = nil
    }
}

class Solution {
    func reverseList(_ head: ListNode?) -> ListNode? {
        var head = head
        var res = ListNode(0)
        var temp: ListNode? = nil
        while head != nil{
            temp = head?.next
            head?.next = res.next
            res.next = head
            head = temp
        }
        return res.next
    }
}

File: 0206/main.go
package main

type ListNode struct {
	Val  int
	Next *ListNode
}

func reverseList(head *ListNode) *ListNode {
	res := &ListNode{0, nil}
	var temp *ListNode = nil
	for head != nil {
		temp = head.Next
		head.Next = res.Next
		res.Next = head
		head = temp
	}
	return res.Next
}
